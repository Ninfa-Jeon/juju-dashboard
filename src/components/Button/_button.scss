.p-button {
  &--filter {
    border-radius: 1rem;
    padding-top: 0;
    padding-bottom: 0.1rem;
  }
}
