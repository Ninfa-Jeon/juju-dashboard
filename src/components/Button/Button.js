import React from "react";

import "./_button.scss";

const Button = props => {
  return <button className="p-button--filter">{props.children}</button>;
};

export default Button;
