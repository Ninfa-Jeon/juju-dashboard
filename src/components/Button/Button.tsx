import { ReactElement } from "react";

import "./_button.scss";

type Props = {
  onClick: () => void;
  children: string;
};

const Button = ({ onClick, children }: Props): ReactElement => {
  return (
    <button className="p-button--filter" onClick={onClick}>
      {children}
    </button>
  );
};

export default Button;
