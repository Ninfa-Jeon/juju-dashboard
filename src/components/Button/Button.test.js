import React from "react";
import { shallow } from "enzyme";

import Button from "./Button";

describe("Button", () => {
  it("renders without crashing and matches snapshot", () => {
    const wrapper = shallow(<Button />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should display the children of button", () => {
    const wrapper = shallow(<Button>Button text</Button>);
    console.log(wrapper.find("button").text());
    expect(wrapper.find("button").text()).toStrictEqual("Button text");
  });
});
