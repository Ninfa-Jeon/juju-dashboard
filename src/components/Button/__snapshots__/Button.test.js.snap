// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders without crashing and matches snapshot 1`] = `
<button
  className="p-button--filter"
/>
`;
